src/solvers/dg_multi/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+  rows, cols = axes(ATr)
+  for i in cols
+    u_i = u[i]
+    du_i = du[i]
+    for j in rows
+      if !skip_index(i,j)
+          du_i += ATr[j,i] * volume_flux(u_i, u[j])
+      end
+    end
+    du[i] = du_i
+  end
+end
+
+# For MultiDG implementations, we construct "physical" differentiation operators by taking linear
+# combinations of reference differentiation operators scaled by geometric change of variables terms.
+# We use LazyArrays.jl for lazy evaluation of physical differentiation operators, so that we can
+# compute linear combinations of differentiation operators on-the-fly in an allocation-free manner.
+function build_lazy_physical_derivative(element::Int, orientation::Int,
+                                        mesh::VertexMappedMesh{2, Tri}, dg, cache)
+  @unpack Qrst_skew_Tr = cache
+  @unpack rxJ, sxJ, ryJ, syJ = mesh.md
+  QrskewTr, QsskewTr = Qrst_skew_Tr
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,element] * QrskewTr + sxJ[1,element] * QsskewTr))
+  else
+    return LazyArray(@~ @. 2 * (ryJ[1,element] * QrskewTr + syJ[1,element] * QsskewTr))
+  end
+end
+
+function build_lazy_physical_derivative(element::Int, orientation::Int,
+                                        mesh::VertexMappedMesh{3, Tet}, dg, cache)
+  @unpack Qrst_skew_Tr = cache
+  QrskewTr, QsskewTr, QtskewTr = Qrst_skew_Tr
+  @unpack rxJ, sxJ, txJ, ryJ, syJ, tyJ, rzJ, szJ, tzJ = mesh.md
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,element]*QrskewTr + sxJ[1,element]*QsskewTr + txJ[1,element]*QtskewTr))
+  elseif orientation == 2
+    return LazyArray(@~ @. 2 * (ryJ[1,element]*QrskewTr + syJ[1,element]*QsskewTr + tyJ[1,element]*QtskewTr))
+  elseif orientation == 3
+    return LazyArray(@~ @. 2 * (rzJ[1,element]*QrskewTr + szJ[1,element]*QsskewTr + tzJ[1,element]*QtskewTr))
+  end
+end
+
+function calc_volume_integral!(du, u::StructArray, volume_integral::VolumeIntegralFluxDifferencing,
+                               mesh::VertexMappedMesh, equations, dg::DG, cache) where {DG <: SBPDGFluxDiff}
+
+  rd = dg.basis
+  @unpack local_values_threaded = cache
+
+  volume_flux_oriented(i) = @inline (u_ll, u_rr)->volume_integral.volume_flux(u_ll, u_rr, i, equations)
+
+  # Todo: simplices. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+  @threaded for e in eachelement(mesh, dg, cache)
+    rhs_local = local_values_threaded[Threads.threadid()]
+    fill!(rhs_local, zero(eltype(rhs_local)))
+    u_local = view(u, :, e)
+    for i in eachdim(mesh)
+      Qi_skew_Tr = build_lazy_physical_derivative(e, i, mesh, dg, cache)
+      hadamard_sum_ATr!(rhs_local, Qi_skew_Tr, volume_flux_oriented(i), u_local)
+    end
+    view(du, :, e) .+= rhs_local ./ rd.wq
+  end
+end
+
+
+function create_cache(mesh::VertexMappedMesh{Dim}, equations, dg::DG,
+                      RealT, uEltype) where {DG <: PolyDGFluxDiff} where {Dim}
```

Comment:
```suggestion
function create_cache(mesh::VertexMappedMesh, equations, dg::PolyDGFluxDiff, RealT, uEltype)
```
instead of explicitly using the parameter, I'd use `ndims(mesh)`. Also, same question regarding the possibility to simplify the dispatch on the `dg` argument as above.